### PR TITLE
go_2-dev: change pname/version to use unstable format

### PR DESCRIPTION
**Pull request by zowoq**
otherwise this causes problems with stable go versions and nix-env

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by mdlayher — COMMENTED** [review]
The change seems reasonable but this appears not to bootstrap in the PR review environment (at least here):

```
$ nixpkgs-review pr 108270
$ git -c fetch.prune=false fetch --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/108270/head:refs/nixpkgs-review/1
remote: Enumerating objects: 6, done.
remote: Counting objects: 100% (6/6), done.
remote: Total 7 (delta 6), reused 6 (delta 6), pack-reused 1
Unpacking objects: 100% (7/7), 642 bytes | 321.00 KiB/s, done.
From ssh://github.com/NixOS/nixpkgs
 + a1e9a94785c...2e7ccd2c2c9 refs/pull/108270/head -> refs/nixpkgs-review/1  (forced update)
$ git worktree add /home/matt/.cache/nixpkgs-review/pr-108270/nixpkgs 712830d101d94ae29a2310d91b4d54c0132f5716
Preparing worktree (detached HEAD 712830d101d)
HEAD is now at 712830d101d Merge pull request #108310 from mdlayher/mdl-bump-corerad
$ git merge --no-commit 2e7ccd2c2c929b05172a33d685b8635211e33eee
Automatic merge went well; stopped before committing as requested
$ nix build --no-link --keep-going --option build-use-sandbox relaxed -f /home/matt/.cache/nixpkgs-review/pr-108270/build.nix
builder for '/nix/store/2yyzgpc9yiv0fkn1qx97naghv99bkh2g-go2-unstable-2020-12-08.drv' failed with exit code 1; last 10 log lines:
  ok          cmd/link/internal/loader        0.091s
  ok     cmd/link/internal/sym   0.083s
  ok     cmd/nm  2.760s
  ok     cmd/objdump     1.792s
  ok     cmd/oldlink/internal/sym        0.052s
  ok     cmd/pack        0.940s
  ok     cmd/trace       0.078s
  ok     cmd/vet 4.452s
  FAIL
  go tool dist: Failed: exit status 1
cannot build derivation '/nix/store/j6xd9pyc8i711h3yzv158w6fjs14ykrd-env.drv': 1 dependencies couldn't be built
[0 built (1 failed), 6 copied (379.7 MiB), 61.2 MiB DL]
error: build of '/nix/store/j6xd9pyc8i711h3yzv158w6fjs14ykrd-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/108270
1 package failed to build:
go_2-dev

```

**Comment by zowoq** [comment]
ofborg doesn't seem to like it either, failed on x86_64-linux here and the original PR.

**Review by kalbasit — APPROVED** [review]

**Comment by fishy** [comment]
Thanks for doing this! After this is merged I'm still getting upgraded from go-1.15.6 to g-2:

```
$ nix-channel --update && nix-env --dry-run -u
unpacking channels...
(dry run; not doing anything)
upgrading 'go-1.15.6' to 'go-2'
these derivations will be built:
  /nix/store/gsx301g781gdc7fx7nnahkmvfsnlcqri-go-2.drv
```

Is there anything else I need to do (or maybe wait for some other release from nixpkgs-channels repo)?

**Comment by zowoq** [comment]
Just need to wait for the channels to update.

**Pull request merged**
